### DISCUSSION TITLE
Shutter for Push-Button

### DISCUSSION THREAD

**Discussion by jwiesel** [arendst/Tasmota#10347] [Ideas]
Hi there,
I installed my first Shelly 2.5 to control my shutters. Great feature of Tasmota! :-)
Unfortunately I am missing something in my specific setup

# My Use Case
* As part of the electrical installation of my shutters push-buttons (German: "Taster") have been used instead of switches (German: "Schalter")
* If I want to open/close the shutter, I have to push the button for several seconds until opening/closing is completed
* I assume those push-buttons have been installed in the past to avoid permanent energy flow from the switch to the shutter motor

# Issue
* After installing the Shelly 2.5 (which is running Tasmota) I realized, if I trigger up/down from the web-interface, the shutters move all the way up/down given the ShutterOpenDuration / ShutterCloseDuration --> Great: that's what I am looking for
* But unfortunaltey I still need to push the physical button (up/down) until the shutters are fully opened or closed. (SetOption80, ShutterMode 1, ShutterOpenDuration 30, ShutterCloseDuration 28)

# Question
* Is it possible to configure tasmota to move my shutters all the way up or down by a short press of my physical push-buttons? (e.g. by simulating the behavior of the web-interface)

**Comment by mklooss** [comment]
had this issues currently... simple trick:
SwitchMode = 3
https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode

**Comment by jwiesel** [comment]
thanks a lot @mklooss !
The following commands worked for a Shelly 2PM with push-buttons.

```
SwitchMode1 3
SwitchMode2 3
```

**Comment by mklooss** [comment]
@jwiesel for Shelly 2.5 you can also use the "Buttons Method":
https://tasmota.github.io/docs/devices/Shelly-2.5/#template

{"NAME":"Shelly 2.5 (buttons)".....
